src/index.js

```diff
@@ -27,36 +27,46 @@ class PluggableJSON {
         let serializer = this.findSerializerForValue(value);
 
         if (serializer) {
-            return serializer.serialize(value);
+            // returns <separator><serializer.type><separator><serialized value>
+            return `${this._separator}${this._escapeSeparators(serializer.type)}${this._separator}${this._escapeSeparators(serializer.serialize(value))}`;
```

Comment:
Stylistically I would find this easier to read without using template literals, and instead just writing :
```
return this._separator + this._escapeSeparators(serializer.type) + 
           this._separator + this._escapeSeparators(serializer.serialize(value));
```